x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForIndexColorStep.java

```diff
@@ -75,17 +75,26 @@ public boolean equals(Object obj) {
 
     @Override
     public Result isConditionMet(Index index, ClusterState clusterState) {
-        String indexName = indexNamePrefix != null ? indexNamePrefix + index.getName() : index.getName();
-        IndexMetadata indexMetadata = clusterState.metadata().index(index);
-
-        if (indexMetadata == null) {
+        // check if the original (managed) index still exists
+        if (clusterState.metadata().index(index) == null) {
             String errorMessage = String.format(Locale.ROOT, "[%s] lifecycle action for index [%s] executed but index no longer exists",
-                getKey().getAction(), indexName);
+                getKey().getAction(), index.getName());
             // Index must have been since deleted
             logger.debug(errorMessage);
             return new Result(false, new Info(errorMessage));
         }
 
+        String indexName = indexNamePrefix != null ? indexNamePrefix + index.getName() : index.getName();
+        IndexMetadata indexMetadata = clusterState.metadata().index(indexName);
+        // check if the (potentially) derived index exists
+        if (indexMetadata == null) {
+            String errorMessage = String.format(Locale.ROOT, "[%s] lifecycle action for index [%s] executed but the target index [%s] " +
+                    "no longer exists",
```

Comment:
I think it's confusing for this to be in past tense since it could be that we checked too early (the index doesn't exist *yet*), so minor suggestion to change it to:

```suggestion
                    "does not exist",
```